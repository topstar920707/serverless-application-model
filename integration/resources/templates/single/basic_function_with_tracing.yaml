Parameters:
  Bucket:
    Type: String
  CodeKey:
    Type: String
  SwaggerKey:
    Type: String
  TracingParamPassThrough:
    Type: String
    Default: PassThrough

Resources:
    ActiveTracingFunction:
        Type: AWS::Serverless::Function
        Properties:
            Handler: index.handler
            Runtime: nodejs12.x
            CodeUri: ${codeuri}
            MemorySize: 128
            Policies:
                - AWSLambdaRole
                - AmazonS3ReadOnlyAccess
            Tracing: Active

    PassThroughTracingFunction:
        Type: AWS::Serverless::Function
        Properties:
            Handler: index.handler
            Runtime: nodejs12.x
            CodeUri: ${codeuri}
            MemorySize: 128
            Policies:
                - AWSLambdaRole
                - AmazonS3ReadOnlyAccess
            Tracing:
              Ref: TracingParamPassThrough
