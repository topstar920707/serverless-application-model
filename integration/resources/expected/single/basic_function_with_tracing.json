[
  { "LogicalResourceId":"ActiveTracingFunction", "ResourceType":"AWS::Lambda::Function"},
  { "LogicalResourceId":"ActiveTracingFunctionRole", "ResourceType":"AWS::IAM::Role"},
  { "LogicalResourceId":"PassThroughTracingFunction", "ResourceType":"AWS::Lambda::Function"},
  { "LogicalResourceId":"PassThroughTracingFunctionRole", "ResourceType":"AWS::IAM::Role" }
]